lib/openactive/helpers/json_ld.rb

```diff
@@ -0,0 +1,100 @@
+module OpenActive
+  module Helpers
+    class JsonLd
+      # The default JSON-LD context for this package.
+      #
+      # @var array
+      @@default_context = [
+        "https://openactive.io/",
+        "https://openactive.io/ns-beta"
+      ].freeze
+
+      # Returns the JSON-LD type from a given thing.
+      #
+      # @param \OpenActive\BaseModel $thing
+      # @return string
+      def self.get_type(thing)
+        # Append "type" attribute for all other classes
+        reflect = ::ReflectionClass.new(thing)
+
+        # Type is just the non-FQ class name
+        reflect.get_short_name
+      end
+
+      # Returns an associative array with the data ready for JSON-LD serialization.
+      #
+      # @param \OpenActive\BaseModel $obj The given instance to convert to JSON-LD
+      # @param object|null $parent The parent node in the structure.
+      # @return array
+      def self.prepare_data_for_serialization(obj, parent = nil)
+        # Get all defined methods for the object
+        # Please note we don't use get_object_vars() here,
+        # As it would only return the public attributes
+        # (BaseModel's are all protected)
+        # class_methods = obj.methods
+
+        data = obj.values
+        #
+        # # Add type to data if not an RpdeItem
+        # if (
+        #     $fq_classname != "\\OpenActive\\Rpde\\RpdeBody" &&
+        #     $fq_classname != "\\OpenActive\\Rpde\\RpdeItem"
+        # ) {
+        #     $data["type"] = self::getType($obj)
+        # }
```

Comment:
Remove commented out PHP code.

Reply:
This is where the RPDE logic slots in. I'll replace this out with a todo comment.